### PR TITLE
[new release] odoc and odoc-parser (2.4.0)

### PR DESCRIPTION
**Pull request by Julow**
OCaml Documentation Generator

- Project page: <a href="https://github.com/ocaml/odoc">https://github.com/ocaml/odoc</a>
- Documentation: <a href="https://ocaml.github.io/odoc/">https://ocaml.github.io/odoc/</a>

### Added

- Add support for external search engines (@panglesd, @EmileTrotignon, ocaml/odoc#972)
  This includes the generation of an index and the display of the results in
  the UI (HTML only).

- Display 'private' keyword for private type extensions (@gpetiot, ocaml/odoc#1019)
- Allow to omit parent type in constructor reference (@panglesd,
  @EmileTrotignon, ocaml/odoc#933)

### Fixed

- Warn and exit when table(s) is not closed (@lubegasimon, ocaml/odoc#1050)
- Hint when list(s) is not closed (@lubegasimon, ocaml/odoc#1050)
- Fix crash on functors returning an alias (@Julow, ocaml/odoc#1046)
- Fix rendering of polymorphic variants (@wikku, @panglesd, ocaml/odoc#971)
- Add references to extension declarations (@gpetiot, @panglesd, ocaml/odoc#949)

### Changed

- Style: Adjusted line height in the TOC to improve readability (@sorawee, ocaml/odoc#1045)
- Style: Remove font fallback to Helvetica, Arial (@Julow, ocaml/odoc#1028)
- Style: Preformatted elements fallback to UA monospace (@toastal, ocaml/odoc#967)
- Style: Sidebar is now stuck to the left of the content instead of the left of
  the viewport (@EmileTrotignon, ocaml/odoc#999)

### CODE REVIEW

**Comment by Julow** [comment]
Odoc's tests are not passing on `< 4.10` and on macos. Is it reasonable to disable the tests for these versions ? Otherwise a patch is available at https://github.com/ocaml/odoc/pull/1061

**Comment by Julow** [comment]
Instead, I chose to redo the release.

File: packages/odoc-parser/odoc-parser.2.4.0/opam [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    "-j"
+    jobs
+    "@install"
+    # Tests are not all associated with a package and would be run if using the
+    # default '@runtest'.
+    "@src/parser/runtest" {with-test}
+  ]
+]
+patches: [
+  patches/0001-Don-t-make-parser-tests-installable.patch
```
Comment (mseri) [inline packages/odoc-parser/odoc-parser.2.4.0/opam:39]:
```suggestion
  "patches/0001-Don-t-make-parser-tests-installable.patch"
```

**Comment by Julow** [comment]
The CI reports a problem with lower bounds on 4.03 but I think it's a problem with crunch's lower bounds. The same test passes in 4.02 and all the other versions.

**Comment by mseri** [comment]
The lower bounds issue is unrelated

**Comment by mseri** [comment]
Thanks

**Pull request merged**